doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+        UInt32 BytesDownloaded;
+        /// DESIGN NOTE: BytesRequired may only be available for downloads done by Windows Package Manager itself.
+        /// Number of bytes required if known
+        UInt32 BytesRequired;
+        /// Download percentage completed
+        UInt32 DownloadPercentage;
+    };
+
+    /// Result of the install
+    runtimeclass InstallResult
+    {
+        InstallResult();
+
+        /// Used by a caller to correlate the install with a caller's data.
+        String CorrelationId{ get; };
+        /// Whether a restart is required to complete the install.
+        Boolean RebootRequired { get; };
+    }
+
+    /// IMPLEMENTATION NOTE: SourceOrigin from AppInstallerRepositorySource.h
+    /// Defines the origin of the app catalog details.
+    enum AppCatalogOrigin
+    {
+        Default,
+        /// User means it was added by the user and could be removed.
+        User,
+        /// Predefined means it came as part of the Windows Package Manager package and cannot be removed.
+        Predefined,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceTrustLevel from AppInstallerRepositorySource.h
+    /// Defines the trust level of the app catalog.
+    enum AppCatalogTrustLevel
+    {
+        None,
+        Trusted,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceDetails from AppInstallerRepositorySource.h
+    /// Interface for retrieving information about an app catalog without acting on it.
+    [version(1)]
+    struct AppCatalogDetails
+    {
+        /// The name of the app catalog.
+        String Name;
+        /// The type of the app catalog.
+        String Type;
+        /// The argument used when adding the app catalog.
+        String Arg;
+        /// The app catalog's extra data string.
+        String Data;
+        /// The app catalog's unique identifier.
+        String Identifier;
+        /// The last time that this app catalog was updated.
+        Windows.Foundation.DateTime LastUpdateTime;
+        /// The origin of the app catalog.
+        AppCatalogOrigin Origin;
+        /// The trust level of the app catalog
+        AppCatalogTrustLevel TrustLevel;
+    };
+
+    /// A metadata item of a package version.
+    enum PackageVersionMetadata
+    {
+        /// The InstallerType of an installed package
+        InstalledType,
+        /// The Scope of an installed package
+        InstalledScope,
+        /// The system path where the package is installed
+        InstalledLocation,
+        /// The standard uninstall command; which may be interactive
+        StandardUninstallCommand,
+        /// An uninstall command that should be non-interactive
+        SilentUninstallCommand,
+        /// The publisher of the package
+        Publisher,
+        /// The locale of the package
+        Locale,
+    };
+
+    /// IMPLEMENTATION NOTE: IPackageVersion from AppInstallerRepositorySearch.h
+    /// A single package version.
+    runtimeclass PackageVersion

```

Comment:
"Version" usually means a specific thing like "6" or "1.2.4.3" - this is a `PackageInfo` or `CatalogPackageInfo` (I can't decide if the types should be prefixed with `Catalog-` yet.)

Reply:
Package:PackageInfo seem like they'd have a 1:1 mapping to me. This has a 1:many mapping. Does PackageVersionInfo work? Or is PackageInfo ok even with the 1:many mapping?